app/models/post.rb

```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  def update_posts_count
+    author.increment!(:posts_counter)
+  end
```

Comment:
- [ ] [optional] @HERMON-1995 great work adding the post-increment method, it would have been better if you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_posts_count method every time a new post has been saved to database, then make the update_posts_count method private. This can apply to comment and likes counter methods as well.